processing/src/main/java/org/apache/druid/segment/ObjectColumnSelector.java

```diff
@@ -19,75 +19,36 @@
 
 package org.apache.druid.segment;
 
+import org.apache.druid.error.DruidException;
+
 /**
- * This class is convenient for implementation of "object-sourcing" {@link ColumnValueSelector}s, it provides default
- * implementations for all {@link ColumnValueSelector}'s methods except {@link #getObject()} and {@link
- * #classOfObject()}.
- *
- * This class should appear ONLY in "extends" clause or anonymous class creation, but NOT in "user" code, where
- * {@link BaseObjectColumnValueSelector} must be used instead.
- *
- * This is a class rather than an interface unlike {@link LongColumnSelector}, {@link FloatColumnSelector} and {@link
- * DoubleColumnSelector} solely in order to make {@link #isNull()} method final.
+ * Restricts selector usage to only allow {@link #getObject()}.
  */
 public abstract class ObjectColumnSelector<T> implements ColumnValueSelector<T>
```

Comment:
I think might have gone a bit overboard in removing the javadoc contract of this class, some of that stuff still seems relevant about only extending it